packages/extension/src/background.js

```diff
@@ -1,3 +1,47 @@
+let currentPopup = null;
+
+function createPopup(hostTabId, service) {
+  function handlePopupClosed(windowId) {
+    if (currentPopup && windowId == currentPopup.id) {
+      currentPopup = null;
+    }
+  }
+
+  try {
+    if (!currentPopup) {
+      const tabIdParam = new URLSearchParams({
+        tabId: hostTabId,
+      });
+      const popUrl = chrome.runtime.getURL(
+        `index.html?${tabIdParam}#/${service.type}`
+      );
+
+      chrome.windows
+        .create({
+          url: popUrl,
+          type: "popup",
+          height: 628,
+          width: 375,
+          left: 1000,
+        })
+        .then((window) => {
+          currentPopup = window;
+          chrome.windows.onRemoved.addListener(handlePopupClosed);
+        });
+    } else {
+      chrome.scripting.executeScript({
+        target: { tabId: hostTabId },
+        function: () =>
+          alert(
+            "Extension is ignoring given request because it can only have one popup at a time. Make sure to close the current popup before making another request"
+          ),
+      });
+    }
+  } catch (error) {
+    console.log("error: ", error);
```

Comment:
Any reason to throw this? Can the app handle gracefully if there is a problem opening the extension?

Reply:
I put that there during debugging and forgot to remove it. It's gone now.